### PR TITLE
add Data.CoList with unfoldr to contrib

### PR DESCRIPTION
**Pull request by Heather**
``` idris
putStrLn $ "unfoldr: " ++ (show
      ( unfoldr
          (λ b → if b ≡ 0 then Nothing else Just (b, b-1)) 10
      )
)

putStrLn $ "unfoldr: " ++ (show
      ( unfoldr
           (λ b → if b ≡ 0 then Nothing else Just (b, b-1)) 30
      )
)
```

``` idris
unfoldr: [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
unfoldr: [30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, ...]
```

### CODE REVIEW

File: libs/contrib/Data/CoList.idr [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Data.CoList
+
+%access public
+
+||| Idris will know that it always can produce a new element in finite time
+codata CoList : Type -> Type
+  where {
```
Comment (david-christiansen) [inline libs/contrib/Data/CoList.idr]: This doesn't match the style used in the rest of the library. Why not

```
codata CoList : Type -> Type where
  Nil : CoList a
  (::) : a -> CoList a -> CoList a
```
?

Comment (Heather) [inline libs/contrib/Data/CoList.idr]: good suggestion

**Comment by david-christiansen** [comment]
It might also be nice to have things like Functor and Show, but that can wait for future contributions as well.

**Comment by Heather** [comment]
@david-christiansen agreed, not sure if that is possible to make total show

**Comment by david-christiansen** [comment]
Lookin' good now!

**Pull request merged**